src/main/java/core/basesyntax/shape/Shape.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.shape;
+
+import core.basesyntax.behaviour.Colorable;
+import core.basesyntax.behaviour.Drawable;
+import core.basesyntax.behaviour.Geometrical;
+import core.basesyntax.type.Color;
+
+public abstract class Shape
+        implements Colorable, Geometrical, Drawable {
+    public static final byte SHAPES_COUNT = 5;
+
+    private Color color;
+
+    public Shape(Color color) {
+        this.color = color;
+    }
+
+    @Override
+    public Color getColor() {
```

Comment:
You can remove 'this' keyword as it is not necessary in this context. Just return 'color' directly.